src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -0,0 +1,466 @@
+                    "{}/{}.templ",
+                    &code_gen_context.language,
+                    code_gen_context.template_name()
+                ),
+                &context,
+            )
+            .unwrap();
+        let mut file = File::create(output.join(format!(
+            "ion_data_model/{}.{}",
+            code_gen_context.language.file_name(&data_model_name),
+            code_gen_context.language.file_extension()
+        )))?;
+        file.write_all(rendered.as_bytes())?;
+        Ok(())
+    }
+
+    /// Maps the given constraint value to a data model
+    fn map_constraint_to_data_model(
+        output: &Path,
+        modules: &mut Vec<String>,
+        code_gen_context: &mut CodeGenContext,
+        tera_fields: &mut Vec<Field>,
+        imports: &mut Vec<Import>,
+        constraint: &IslConstraint,
+    ) -> CodeGenResult<()> {
+        match constraint.constraint() {
+            IslConstraintValue::Element(isl_type, _) => {
+                Self::verify_data_model_consistency(code_gen_context, DataModel::SequenceStruct)?;
+                Self::generate_struct_field(
+                    code_gen_context,
+                    tera_fields,
+                    isl_type,
+                    output,
+                    modules,
+                    "value",
+                    imports,
+                )?;
+            }
+            IslConstraintValue::Fields(fields, _content_closed) => {
+                Self::verify_data_model_consistency(code_gen_context, DataModel::Struct)?;
+                for (name, value) in fields.iter() {
+                    Self::generate_struct_field(
+                        code_gen_context,
+                        tera_fields,
+                        value.type_reference(),
+                        output,
+                        modules,
+                        name,
+                        imports,
+                    )?;
+                }
+            }
+            IslConstraintValue::Type(isl_type) => {
+                Self::verify_data_model_consistency(code_gen_context, DataModel::UnitStruct)?;
+                Self::generate_struct_field(
+                    code_gen_context,
+                    tera_fields,
+                    isl_type,
+                    output,
+                    modules,
+                    "value",
+                    imports,
+                )?;
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    /// Verify that the current data model is same as previously determined data model
+    fn verify_data_model_consistency(
+        code_gen_context: &mut CodeGenContext,
+        current_data_model: DataModel,
+    ) -> CodeGenResult<()> {
+        if let Some(data_model) = &code_gen_context.data_model {
+            if data_model != &current_data_model {
+                return invalid_data_model_error("Can not determine data model, constraints are mapping to different data models.");
+            }
```

Comment:
This error message requires some understanding of the code generator's internals; we should aim to describe the problem in terms of ISL since that's what the user is providing.
```
Constraint {current_constraint:?} (for {abstract_data_type}) conflicts with prior constraints: {constraints_we_already_evaluated:?} (for {conflicting_abstract_data_type}.
```

Reply:
Thats a great suggestion. Thanks!

Reply:
This requires a `Display` implementation of constraints in `ion-schema`.  Reference issue https://github.com/amazon-ion/ion-schema-rust/issues/204